### ISSUE TITLE
Tips for solving readonly missing feature: link unavailable and so on

### ISSUE THREAD

**Issue by hh54188** [tinymce/tinymce#4575]
**Do you want to request a *feature* or report a *bug*?**

Neither

**What is the current behavior?**

In `readonly` mode, I found some features are missing (or bug ?):

* Text can not be selected
* Table can be adjusted
* Link is unavailable

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in 
previous versions of TinyMCE?**

4.8.2 (2018-08-09)

**My solution**

Because the editor's content saved as html segment, so when I need the `readonly` mode, I would create an iframe dom node and put the html segment into it.

Here is my code:

```javascript
  renderReportPreview = contentHtml => {
    const iframe = document.querySelector('iframe[name=preview]')
    if (iframe) {
      const cssLink = document.createElement('link')
      // cssLink.href = 'https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css'
      // I prefer semantic-ui, semantic-ui is more like tinyMce style
      cssLink.href = 'https://cdn.bootcss.com/semantic-ui/2.3.1/semantic.min.css'
      cssLink.rel = 'stylesheet'
      cssLink.type = 'text/css'
      iframe.contentWindow.document.head.appendChild(cssLink)
      // I escape the origin editor's content, so I need decode them back
      iframe.contentWindow.document.body.innerHTML = htmlDecode(contentHtml)
      const allTables = iframe.contentWindow.document.body.querySelectorAll(
        'table'
      )
      Array.from(allTables).forEach(table => {
        // ui celled table for compatible semantic-ui
        // table table-bordered for compatible bootstrap
        table.className = 'ui celled table table-bordered'
      })
      this.setState({
        previewRendered: true,
      })
    }
  }
``` 

As the code description above, I also custom my theme for editor's content, by using `bootstrap` style or `semantic-ui` style. I prefer `semantic-ui` because I think it is more like. The way using them is also easy, just inject an style link into the iframe document

So when you need `readonly` mode, or a preview for editor's content. You may take this way to avoid native `readonly` missing features.

**Comment by TheSpyder** [comment]
This probably broke in 4.7.12, we are tracking an updated fix for #4249 and will update this ticket when it's merged.

**Comment by wonderbugtiny** [comment]
This has been fixed in 5.0.6. Please let us know if you find any further problems. Thank you!
